### PR TITLE
libbpf-tools/offcputime: Use CO-RE operation to read kernel data

### PR DESCRIPTION
**Pull request by ekyooo**
Use CO-RE API to read the value of a field from task_struct.

Signed-off-by: Eunseon Lee <es.lee@lge.com>

### CODE REVIEW

File: libbpf-tools/offcputime.bpf.c [hunk 1]
```diff
@@ -66,15 +66,15 @@ int BPF_PROG(sched_switch, bool preempt, struct task_struct *prev, struct task_s
 	u32 pid;
 
 	if (allow_record(prev)) {
-		pid = prev->pid;
+		pid = BPF_CORE_READ(prev, pid);
```
Comment (chenhengqi) [inline libbpf-tools/offcputime.bpf.c:69]: `tp_btf` is BTF-awared, pointer chasing should work, right ?

Comment (ekyooo) [reply libbpf-tools/offcputime.bpf.c:69]: Oh... I didn't know that. Thank you for the information. Does `tp_btf` support depend on kernel version?
A segfault occurred when running the default version of offfcputime on a Linux v5.4 kernel environment. So I switched tp_btf to tracepoint as below. So this patch was needed.
```
-SEC("tp_btf/sched_switch")
+SEC("tracepoint/sched/sched_switch")
```
Could you please tell me if `tp_btf` is supported in Linux v5.4 as well?
Thank you in advance.

Comment (ekyooo) [reply libbpf-tools/offcputime.bpf.c:69]: Ok. I got information from another engineer that tp_btf is not supported in Linux v5.4. I abandon this patch. Thank you.

Comment (chenhengqi) [reply libbpf-tools/offcputime.bpf.c:69]: See https://github.com/iovisor/bcc/issues/4231.

Comment (ekyooo) [reply libbpf-tools/offcputime.bpf.c:69]: Oh..!! Thank you. I'll read it carefully.

**Pull request closed without merging**